### ISSUE TITLE
[REVIEW]: Nempy: A Python package for modelling the Australian National Electricity Market dispatch procedure

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3596]
**Submitting author:** @nick-gorman (<a href="http://orcid.org/0000-0002-6505-761X">Nicholas Gorman</a>)
**Repository:** <a href="https://github.com/UNSW-CEEM/nempy" target ="_blank">https://github.com/UNSW-CEEM/nempy</a>
**Version:** v1.1.2
**Editor:** @timtroendle
**Reviewer:** @noah80, @robinroche
**Archive:** <a href="https://doi.org/10.5281/zenodo.5989170" target="_blank">10.5281/zenodo.5989170</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e/status.svg)](https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e"><img src="https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e/status.svg)](https://joss.theoj.org/papers/70ac80e2603b4eef6157d9d783ac0f8e)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/UNSW-CEEM/nempy" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@noah80 & @robinroche, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @timtroendle know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

## Review checklist for @noah80
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/UNSW-CEEM/nempy">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/UNSW-CEEM/nempy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nick-gorman) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

## Review checklist for @robinroche
✨ Important: Please do not use the _Convert to issue_ functionality when working through this checklist, instead, please open any new issues associated with your review <a target="_blank" href="https://github.com/UNSW-CEEM/nempy">in the software repository associated with the submission</a>. ✨

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/UNSW-CEEM/nempy">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nick-gorman) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @noah80, @robinroche it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3596 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.65 s (98.5 files/s, 179070.6 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
XML                              2              4              0          93528
Python                          46           3347           7224           5812
SVG                              2              0              0           5223
HTML                             1             91              0            365
Markdown                         1             47              0            264
reStructuredText                 8             98             92            201
YAML                             1              0              2             51
DOS Batch                        1              8              1             26
make                             1              4              7              9
INI                              1              0              0              3
-------------------------------------------------------------------------------
SUM:                            64           3599           7326         105482
-------------------------------------------------------------------------------


Statistical information for the repository '6a58250476f57617e864f97d' was
gathered on 2021/08/10.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Ben Elliston                     1            10             39            0.10
Nicholas Gorman                  2           459            150            1.23
Nick                            26          8124           6965           30.38
Nick Gorman                      2           356            505            1.73
nick-gorman                    111         24034           9032           66.57

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Ben Elliston                  3           30.0          1.2                0.00
nick-gorman               16380           68.2          3.1                6.53
```

**Comment by timtroendle** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by timtroendle** [comment]
@noah80, @robinroche, I'll add a reminder for both of you to notify you when half of the six weeks has elapsed.

**Comment by timtroendle** [comment]
@whedon remind @noah80 in three weeks

**Comment by whedon** [comment]
Reminder set for @noah80 in three weeks

**Comment by timtroendle** [comment]
@whedon remind @robinroche in three weeks

**Comment by whedon** [comment]
Reminder set for @robinroche in three weeks

**Comment by timtroendle** [comment]
@robinroche, @noah80, I will be out of office for the duration mentioned above. If there are any questions from your side about this review, it would be great if you could ask them before I leave.

**Comment by whedon** [comment]
:wave: @robinroche, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @noah80, please update us on how your review is going (this is an automated reminder).

**Comment by robinroche** [comment]
>  @robinroche, please update us on how your review is going (this is an automated reminder).

I have started the review, and will complete it in the coming weeks.

**Comment by timtroendle** [comment]
@noah80, can you please let us know how your review is going?

**Comment by whedon** [comment]
:wave: @noah80, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @robinroche, please update us on how your review is going (this is an automated reminder).

**Comment by robinroche** [comment]
@timtroendle 
@nick-gorman

Thanks for this interesting submission. I installed the package and ran the examples without trouble.

After reviewing the submission, here are some suggestions for improvement:
1. While I understand the general purpose of the code, additional information on how it could be used could be useful. For example: provide use cases, questions it could help answer, etc.
2. Briefly introduce the basics of electricity markets, e.g., a very simplified version of a market clearing problem, its objectives and constraints. Similarly, introduce acronyms such as FCAS.
3. Explain how to interpret example results. What should be expected and how to know it worked? Could graphs be added to have a more visual result?
4. Specify the size of the data to download in the description of example 2 (this is only given in the comments). As this is really a lot of data, could it be possible to design another example with less data requirements?
5. Briefly explain the main steps of the dispatch procedure.
6. Be more specific when you mention: "Significant changes to the dispatch process are also likely to occur soon"
7. The text mentions "Version 1 will be a stable release and ongoing minor updates or patches will remain backwards compatible." The current version is 1.0.1
8. Link to a document explaining how the actual NEM works.
9. Explain how to use solvers and what criteria they should meet.
10. List "todos" you intend to implement in the future, possibly with priorities.
11. Move the long list of examples closer to where they would make sense, add more details in the description
12. Explain how to check the computation time.
13. Explain how to report an issue or get support.
14. List the dependencies.
15. Regarding the paper abstract, make it simpler for a wider audience and clarify the problems it can help solve and/or the questions it can help answer. Who is the target audience?

**Comment by timtroendle** [comment]
@noah80, can you please give us an indication about the progress of your review?

**Comment by noah80** [comment]
@timtroendle 
@nick-gorman

Sorry for the delay. After reviewing the software too, I reached similar conclusions to Robin.

1. The code is well-documented and readable.
2. Even after reading the documentation and the paper I still don't know what the software could be used for. Why would anyone want to model the dispatch?
3. The predefined examples show how to use the software, but don't explain the results. No graphs are generated. Could those be added to help the users better understand the results? 
4. The research question for the software is not explained. Maybe add some other sources that explain why this software is used and what results can be generated with the software?

Right now with the given software it is rather difficult to judge if the software delivers the promised functionality and who could use the software for what purpose.

**Comment by nick-gorman** [comment]
@robinroche and @noah80 thank you both for your reviews, I really appreciate the feedback. I'll aim to address your comments and questions in the next couple of weeks.

**Comment by nick-gorman** [comment]
Hi @robinroche and @noah, thanks again for reviewing and your help comments. I've drafted a new version of the paper, made some changes to the documentation, and released a new version of the software with a minor update to try and address them.

Bellow I've tried highlight specifically where and how I've address each of your comments, please ask me to clarify if needed.

## @robinroche comments:

1. While I understand the general purpose of the code, additional information on how it could be used could be useful. For example: provide use cases, questions it could help answer, etc.

I've added significantly to the statement of need section, lines 60 - 117. I've outlined 4 distinct use cases, and provided a simple example analysis that briefly demonstrates one of the use cases.

2. Briefly introduce the basics of electricity markets, e.g., a very simplified version of a market clearing problem, its objectives and constraints. Similarly, introduce acronyms such as FCAS

I've edited the summary section of the paper to address this, lines 10 - 13. I've introduced electricity markets, the way the NEM spot market works simply, and how it needs more complex features like FCAS in real life.

3. Explain how to interpret example results. What should be expected and how to know it worked? Could graphs be added to have a more visual result?

I have edited the examples in the documentation to address this. For examples 1 - 5 I have added additional inline comments to explain the results. For examples 6 and 7 I've added graphs to make interpreting the results easier.

https://nempy.readthedocs.io/en/latest/examples.html#examples

4. Specify the size of the data to download in the description of example 2 (this is only given in the comments). As this is really a lot of data, could it be possible to design another example with less data requirements?

For the 2nd example in the read me and examples 6 - 8 in the documentation I've added in text warnings about the volume of data to be downloaded.

I've also published a new version of the software that allows the historical data xml cache to be built on a daily rather than monthly basis. This reduces the data downloaded for the examples from 65GB to 8.5GB.

https://github.com/UNSW-CEEM/nempy/releases/tag/v1.1.0

5. Briefly explain the main steps of the dispatch procedure.

I've added a section to the documentation to outline the steps of the dispatch procedure.

https://nempy.readthedocs.io/en/latest/intro.html#dispatch-procedure-outline

6. Be more specific when you mention: "Significant changes to the dispatch process are also likely to occur soon"

I've added additional detail: 

"As part of the ongoing reform process significant changes to the dispatch process have also been proposed, for example, the introduction of an operating reserve market is being considered, as is the adoption of a dynamic intra-regional loss model."

https://nempy.readthedocs.io/en/latest/intro.html#introduction

7. The text mentions "Version 1 will be a stable release and ongoing minor updates or patches will remain backwards compatible." The current version is 1.0.1

This is standard semantic versioning, it's still the major version 1, the second and third numbers just get updated to indicate minor updates (but backwards compatible) and patches respectively.

I've also tried to update the text to make this clearer without being verbose in the intro.

"Version 1 will be a stable release and ongoing minor updates or patches will remain backwards compatible. Currently the latest minor release is v1.1.0."

8. Link to a document explaining how the actual NEM works.

I've added links to a description of the NEM in both the readme and the readthedocs page:

https://github.com/UNSW-CEEM/nempy#introduction
https://nempy.readthedocs.io/en/latest/intro.html#introduction

Also, the fourth use case in the paper now explains that there is no complete description of the dispatch process provided by AEMO, and references several documents addressing aspects of the dispatch process. 

9. Explain how to use solvers and what criteria they should meet.

The Run-time section (https://nempy.readthedocs.io/en/latest/intro.html#run-time) now links to the reference documentation on solvers.

10. List "todos" you intend to implement in the future, possibly with priorities.

The Ongoing Work section (updated) provides this information

https://nempy.readthedocs.io/en/latest/intro.html#ongoing-work

11. Move the long list of examples closer to where they would make sense, add more details in the description

Not sure exactly what you mean by moving the examples, there was an error where they were showing up in the top level menu, is this what you meant?

Some more detail has been added to the descriptions.

12. Explain how to check the computation time.

A brief explanation has been added.

"The total runtime was calculated using the python time module and measuring the time taken from the loading of inputs to the extraction of results from the model. The runtime of different sub-process, i.e. loading of the XML file, was measured by inserting timing code into the Nempy source code where required."

https://nempy.readthedocs.io/en/latest/intro.html#run-time

13. Explain how to report an issue or get support.

This is explained in the readme on GitHub, I've added links to the the issues register and discussion board to make this clearer.

https://github.com/UNSW-CEEM/nempy#community

14. List the dependencies.

Added this section to the documentation

https://nempy.readthedocs.io/en/latest/intro.html#dependencies

15. Regarding the paper abstract, make it simpler for a wider audience and clarify the problems it can help solve and/or the questions it can help answer. Who is the target audience?

Lines 10 - 17 of the paper should now provide a simpler intro for the wider audience. Lines 17 - 26 aim to softly introduce some of the additional complexity. Lines 60 - 117 deal specifically with use cases and example question. Lines 61 and 62 say who the target audience is.

I've also added a section to the documentation that talks about target users and uses cases:
https://nempy.readthedocs.io/en/latest/intro.html#example-use-cases 

## @noah80 comments:

1. The code is well-documented and readable.

2. Even after reading the documentation and the paper I still don't know what the software could be used for. Why would anyone want to model the dispatch?

I've added significantly to the statement of need section, lines 60 - 117. I've outlined 4 distinct use cases, and provided a simple example analysis that briefly demonstrates one of the use cases.

3. The predefined examples show how to use the software, but don't explain the results. No graphs are generated. Could those be added to help the users better understand the results?

I have edited the examples in the documentation to address this. For examples 1 - 5 I have added additional inline comments to explain the results. For examples 6 and 7 I've added graphs to make interpreting the results easier.

https://nempy.readthedocs.io/en/latest/examples.html#examples

4. The research question for the software is not explained. Maybe add some other sources that explain why this software is used and what results can be generated with the software?

As for comment two I think lines 60 - 177 of the paper should now address this. I've also added a section on use cases to the documentation

https://nempy.readthedocs.io/en/latest/intro.html#example-use-cases

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by timtroendle** [comment]
👋 @noah80 we are waiting for your update on this review. Can you let us know by when you will be able to update this?

**Comment by timtroendle** [comment]
@noah80, this review is currently blocked by the unfinished second review. Can you please give us an update?

**Comment by noah80** [comment]
The PDF update is excellent and fulfills now all criteria. I couldn't get the tests running so far though. 

@nick-gorman : Could you contact me? Maybe I'm doing something wrong.

**Comment by nick-gorman** [comment]
> The PDF update is excellent and fulfills now all criteria. I couldn't get the tests running so far though.
> 
> @nick-gorman : Could you contact me? Maybe I'm doing something wrong.

@noah80 Good to hear I've been able to address your comments.

I've opened an issue on the nempy repo regarding the unit tests, see https://github.com/UNSW-CEEM/nempy/issues/4 . Can you reply on this issue and provide details of your problem running the unit tests?

I don't have your email, but you can contact me at n.gorman - at - unsw.edu.au

**Comment by nick-gorman** [comment]
@noah80 please let me know if UNSW-CEEM/nempy#4 and UNSW-CEEM/nempy#5 have resolved your problems.

**Comment by noah80** [comment]
Still having some issues with running the examples in a clean enviroment due to missing dependencies in the package. This is not critical for the functionality but bad for the usability of your software in my opinion and I would suggest fixing this.

**Comment by nick-gorman** [comment]
> Still having some issues with running the examples in a clean enviroment due to missing dependencies in the package. This is not critical for the functionality but bad for the usability of your software in my opinion and I would suggest fixing this.

@noah80, good point, I've tried to address as per UNSW-CEEM/nempy#5

**Comment by noah80** [comment]
I was able to run all examples now after manually installing the missing dependencies. 

While it would be nice to automate that part or surround make the code sections in the examples with try/except, it all works and looks usable. 

@timtroendle All my issues have been adressed. 

@nick-gorman Thanks for the cooperation!

**Comment by timtroendle** [comment]
@whedon check references

**Comment by timtroendle** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by timtroendle** [comment]
@whedon  check references from branch joss-paper

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.3509134 is OK
-  10.25080/Majora-92bf1922-00a  is OK
- 10.1038/s41586-020-2649-2 is OK

MISSING DOIs

- 10.1016/j.rser.2018.08.002 may be a valid DOI for title: A review of modelling tools for energy and electricity systems with large shares of variable renewables
- 10.1016/j.apenergy.2021.116731 may be a valid DOI for title: Trends in tools and approaches for modelling the energy transition
- 10.1016/j.rser.2020.110195 may be a valid DOI for title: A systemic approach to analyze integrated energy system modeling tools: A review of national models
- 10.1038/sdata.2018.203 may be a valid DOI for title: Open grid model of Australia’s National Electricity Market allowing backtesting against historic data
- 10.1016/j.enpol.2013.01.052 may be a valid DOI for title: Retrospective modeling of the merit-order effect on wholesale electricity prices from distributed photovoltaic generation in the Australian National Electricity Market

INVALID DOIs

- None
```

**Comment by timtroendle** [comment]
Thanks a lot @noah80 and @robinroche for your reviews!

We can now move on with the acceptance process. I had a quick read of the paper and found three small issues:

1. There is a typo in line 15: "the end of 5 the minute dispatch".
2. There are typos in line 2 and 3 of the legend in figure 1: "gernic constraints".
3. Whedon spotted 5 missing DOIs.

@nick-gorman , could you at this point please:
- [ ] Fix above-mentioned issues, where possible
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by nick-gorman** [comment]
Thanks @timtroendle, @noah80 and @robinroche

I'm away from my work computer, which has the chart files that need correcting. @timtroendle I'll do the final tasks in a couple of weeks when I'm back at work.

**Comment by nick-gorman** [comment]
@timtroendle

I've been able to fix the issues you found.

The release tag of the software is v1.1.2

And the zendo doi is: 10.5281/zenodo.5989170

**Comment by timtroendle** [comment]
@whedon generate pdf from branch joss-paper

**Comment by timtroendle** [comment]
@whedon check references from branch joss-paper

**Comment by whedon** [comment]
```
Attempting to check references... from custom branch joss-paper
```

**Comment by timtroendle** [comment]
@whedon set 10.5281/zenodo.5989170 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5989170" target="_blank">10.5281/zenodo.5989170</a> is the archive.

**Comment by timtroendle** [comment]
@whedon set v1.1.2 as version

**Comment by timtroendle** [comment]
Thanks @nick-gorman. It all looks good and I will now recommend your submission for acceptance.

**Comment by timtroendle** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2932

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2932, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.5281/zenodo.3509134 is OK
-  10.25080/Majora-92bf1922-00a  is OK
- 10.1038/s41586-020-2649-2 is OK
- 10.1016/j.rser.2018.08.002 is OK
- 10.1016/j.apenergy.2021.116731 is OK
- 10.1016/j.rser.2020.110195 is OK
- 10.1038/sdata.2018.203 is OK
- 10.1016/j.enpol.2013.01.052 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kthyng** [comment]
Hi @nick-gorman! Items for wrapping up:

1. I see your Zenodo archive and looks good, except there seems to be a "2" in the middle of your title?
2. Please check the capitalization in your references. For example, Grozev et al has "australia" uncapitalized. You can preserve capitalization by putting {} around characters or strings in your .bib file.

**Comment by nick-gorman** [comment]
Thanks @kthyng 

Not sure how that 2 got there, or how I didn't see it. Good tip with the curly brackets.

These issues should be resolved.

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@whedon generate pdf from branch joss-paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss-paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@nick-gorman ok everything looks good! But, do you want me to  accept this submission from the `joss-paper` branch or were you going to merge it first? Either way is fine, I just need to know which.

**Comment by nick-gorman** [comment]
@kthyng I've just merged joss-paper into master, so should be good to go from master

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03596/joss.03596/10.21105.joss.03596.pdf) :page_facing_up: :point_left:

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1491618526934749184) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2938
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03596](https://doi.org/10.21105/joss.03596)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congratulations on your new publication @nick-gorman! Many thanks to editor @timtroendle and reviewers @noah80 and @robinroche for your time, hard work, and expertise!!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03596/status.svg)](https://doi.org/10.21105/joss.03596)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03596">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03596/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03596/status.svg
   :target: https://doi.org/10.21105/joss.03596
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03596/status.svg)](https://doi.org/10.21105/joss.03596)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by nick-gorman** [comment]
Thank you @noah80 @robinroche @timtroendle and @kthyng!
